### PR TITLE
Run linter during CI

### PR DESCRIPTION
**Pull request by fiunchinho**
- [X] Change is code complete and matches issue description.
- [X] Change is covered by existing or new tests.

#### Description
Run the linter (golint and go vet) during CI, so we now the formatting of the code is correct.

#### Special notes for the reviewer(s)
Heads up: I've cleaned some unused stuff from the Makefile, if you think it should be kept, just let me know.

### CODE REVIEW

**Comment by fiunchinho** [comment]
Running the linter locally, I'm getting the following error:
```
LINTING
/Users/jose.armesto/dev/src/github.com/jenkins-x/jx/pkg/auth/user_auth_test.go is in package auth_test, not auth
make: *** [github.com/jenkins-x/jx/pkg/auth] Error 1
```

I'm no go expert. Anyone know how to fix this?

File: Makefile [hunk 1]
```diff
@@ -325,33 +319,19 @@ $(FGT):
...
-	done ; exit $$ret
+lint: $(GOLINT)
+	@echo "--> linting code with 'go lint' tool"
+	$(GOLINT) -min_confidence 1.1 ./...
 
 .PHONY: vet
 vet: tools.govet
 	@echo "--> checking code correctness with 'go vet' tool"
-	@go vet ./...
+	@go vet ./... || true
```
Comment (jglick) [inline Makefile]: ?

File: pkg/apis/jenkins.io/v1/types_extensions.go [hunk 2]
```diff
@@ -97,7 +97,7 @@ type ExtensionExecution struct {
 	Name                 string                `json:"name,omitempty"  protobuf:"bytes,1,opt,name=name"`
 	Description          string                `json:"description,omitempty"  protobuf:"bytes,2,opt,name=description"`
 	Script               string                `json:"script,omitempty"  protobuf:"bytes,3,opt,name=script"`
-	EnvironmentVariables []EnvironmentVariable `json:"environmentVariables,omitempty protobuf:"bytes,4,opt,name=environmentvariables"`
+	EnvironmentVariables []EnvironmentVariable `json:"environmentVariables,omitempty" protobuf:"bytes,4,opt,name=environmentvariables"`
```
Comment (jglick) [inline pkg/apis/jenkins.io/v1/types_extensions.go]: Oops!

File: Makefile (hunk 1 above)
Comment (fiunchinho) [reply Makefile]: I figured we could send another PR to fix `go vet` issues, and keep the build not failing for now (like it was).

**Comment by pmuir** [comment]
flaky test

**Comment by pmuir** [comment]
/test
/hold
Holding PRs as we have a big refactor to get in first

**Comment by fiunchinho** [comment]
It looks like the build has ended successfully but somehow it didn't report its status back.

**Pull request merged**